File: GeneratedEntityReaderClasses/Components/PixelSceneComponent.cs
namespace GeneratedNoitaClasses;
public class PixelSceneComponent : NoitaComponentBase,  iNoitaType<PixelSceneComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaString pixel_scene { get; set; } = new();
	public NoitaString pixel_scene_visual { get; set; } = new();
	public NoitaString pixel_scene_background { get; set; } = new();
	public NoitaInt background_z_index { get; set; } = new();
	public NoitaFloat offset_x { get; set; } = new();
	public NoitaFloat offset_y { get; set; } = new();
	public NoitaBool skip_biome_checks { get; set; } = new();
	public NoitaBool skip_edge_textures { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		pixel_scene.Read(s);
		pixel_scene_visual.Read(s);
		pixel_scene_background.Read(s);
		background_z_index.Read(s);
		offset_x.Read(s);
		offset_y.Read(s);
		skip_biome_checks.Read(s);
		skip_edge_textures.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		pixel_scene.Write(s);
		pixel_scene_visual.Write(s);
		pixel_scene_background.Write(s);
		background_z_index.Write(s);
		offset_x.Write(s);
		offset_y.Write(s);
		skip_biome_checks.Write(s);
		skip_edge_textures.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/PixelSceneComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class PixelSceneComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    pixel_scene:NoitaString = field(default_factory=NoitaString)
    pixel_scene_visual:NoitaString = field(default_factory=NoitaString)
    pixel_scene_background:NoitaString = field(default_factory=NoitaString)
    background_z_index:NoitaInt = field(default_factory=NoitaInt)
    offset_x:NoitaFloat = field(default_factory=NoitaFloat)
    offset_y:NoitaFloat = field(default_factory=NoitaFloat)
    skip_biome_checks:NoitaBool = field(default_factory=NoitaBool)
    skip_edge_textures:NoitaBool = field(default_factory=NoitaBool)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.pixel_scene.Read(s)
        self.pixel_scene_visual.Read(s)
        self.pixel_scene_background.Read(s)
        self.background_z_index.Read(s)
        self.offset_x.Read(s)
        self.offset_y.Read(s)
        self.skip_biome_checks.Read(s)
        self.skip_edge_textures.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.pixel_scene.Write(s)
        self.pixel_scene_visual.Write(s)
        self.pixel_scene_background.Write(s)
        self.background_z_index.Write(s)
        self.offset_x.Write(s)
        self.offset_y.Write(s)
        self.skip_biome_checks.Write(s)
        self.skip_edge_textures.Write(s)

